### ISSUE TITLE
Feature Request - Deal_stage modeling

### ISSUE THREAD

**Issue by fivetran-kristin** [fivetran/dbt_hubspot#30]
This will be added to the next release of Hubspot:

Model DEAL_STAGE (which has History Mode) in both Hubspot source & transformation packages.

**Comment by Spince** [comment]
@fivetran-kristin any idea when the next release is planned?

**Comment by fivetran-kristin** [comment]
@Spince thanks for the ping here. We're actually in the middle of a 2-week package maintenance sprint and this is on our list. 

I do have a question for you -- What deal_stage related metrics do you want to see in the transform package?

**Comment by Spince** [comment]
@fivetran-kristin thank you! our analysts are giving this some thought (and familiarizing themselves with the existing transformed models as they're new for us). should have a response by EOD tomorrow

**Comment by raphshin** [comment]
Hi @fivetran-kristin - following up on the last comment by @Spince. Please see the wishlist of fields for DEAL_STAGE: 

deal_stage_id
deal_stage_name
deal_pipeline_id
deal_pipeline_name
company_id
company_name
timestamp: entering the specific stage_id
timestamp: exiting/moving to different stage_id
stageisactive
piplineisactive
deal_owner_id
deal_owner_name

I have a question though. What is the main differentiator between DEAL_HISTORY vs DEAL_STAGE?

**Comment by kaufmannie** [comment]
@fivetran-kristin - would it be helpful if a create an issue to add deal_stages model on the dbt_hubspot_source repo as well? or is that just duplicative?

**Comment by fivetran-joemarkiewicz** [comment]
Hi @kaufmannie I am currently working on adding some Service Hub staging models to the source package so I can easily include the deal_stage model as well to work with this release!

@raphshin Thank you for your wishlist! I'll work on getting these integrated soon. Stay tuned for another message from me when I have a working branch so you may test it out before release if you would like to provide additional feedback.

**Comment by fivetran-joemarkiewicz** [comment]
Hi @kaufmannie , @raphshin , and @Spince!

Good news, we were able to complete the working branch with the `deal_stage` updates you requested for both the source and transform packages. If you have time we would greatly appreciate your feedback and input on these changes prior to us merging and releasing the new version. You may update the hubspot dependency within your `packages.yml` to the below config:
```yml
packages:
  - git: https://github.com/fivetran/dbt_hubspot_source.git
    revision: deal_stage-updates
    warn-unpinned: false
```

A few things to point out is the source package included other updates as well to allow for pass through fields with more flexibility on the end user side. I would recommend taking a look at the documentation updates prior to executing the models.

Additionally, we discovered within Issue #35 that deal to company is a 1:n relationship and can fan out the data if those relationships exists. Therefore we decided to leave the company details in your wishlist out of the final model as it could cause schema test failures for other users. You can easily still join the company data to the deal_stage data if you would like in a separate query.

Let me know what your feedback is and how the package updates look. Thanks!

**Comment by fivetran-joemarkiewicz** [comment]
Hi @kaufmannie , @raphshin , and @Spince 

We have since merged the changes from PR #37 to master and cut a new `v0.3.2` release of the dbt_hubspot package. You should see this new release live on the [dbt hub](https://hub.getdbt.com/) most likely by Monday!

With the addition of the `hubspot__deal_stages` model, this Issue can be closed. If you have additional questions related to this Issue please feel free to reopen the issue and continue the conversation. If you have any other questions or requests for this package, please feel free to open a new Issue. Thank you!
